### ISSUE TITLE
[java] JUnit4TestShouldUseTestAnnotation false positive with TestNG

### ISSUE THREAD

**Issue by krdabrowski** [pmd/pmd#4634]
**Affects PMD Version:** 
7.0.0-rc3

**Rule:**
[JUnit4TestShouldUseTestAnnotation](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#junit4testshouldusetestannotation)

**Description:**
Rule complaints about missing `@Test` annotation even if it is present, but from `org.testng.annotations` package.
Probably TestNG `org.testng.annotations.Test` annotation is missing in rule definition: https://github.com/pmd/pmd/blob/3e2de671907a2b2984880e50105072fd6c26116c/pmd-java/src/main/resources/category/java/bestpractices.xml#L783C75-L783C75

**Code Sample demonstrating the issue:**

```java
import org.testng.annotations.Test;
public class A {
  @Test
  void test() {}  // report warning
}
```

**Expected outcome:**
PMD reports a violation: `JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest` at line 4, but that's wrong - `@Test` annotation is present by the test method.

**Comment by adangel** [comment]
See also #4532
